src/js/test/unit/common/contextManager.test.ts

```diff
@@ -0,0 +1,212 @@
+import * as chai from "chai";
+import chaiAsPromised from "chai-as-promised";
+
+chai.use(chaiAsPromised);
+const assert = chai.assert;
+const expect = chai.expect;
+
+import {
+  withContext,
+  createContextWrapper,
+} from "../../../common/contextManager";
+
+describe("withContext", () => {
+  it("synchronous execution", () => {
```

Comment:
Can you add tests for when setup and cleanup raise? Synchronously and asynchronously?